FluentAssertions.Core/Primitives/ObjectAssertions.cs

```diff
@@ -63,6 +63,58 @@ public AndConstraint<ObjectAssertions> NotBe(object unexpected, string because =
         }
 
         /// <summary>
+        /// Asserts that an object is an enum and has a specified flag
+        /// </summary>
+        /// <param name="expectedFlag">The expected flag.</param>
+        /// <param name="because">
+        /// A formatted phrase explaining why the assertion should be satisfied. If the phrase does not 
+        /// start with the word <i>because</i>, it is prepended to the message.
+        /// </param>
+        /// <param name="reasonArgs">
+        /// Zero or more values to use for filling in any <see cref="string.Format(string,object[])" /> compatible placeholders.
+        /// </param>
+        public AndConstraint<ObjectAssertions> HaveFlag(Enum expectedFlag, string because = "", 
+            params object[] reasonArgs)
+        {
+            Enum subject = Subject as Enum;
+
+            Subject.Should().BeOfType(expectedFlag.GetType(), because, reasonArgs);
+
+            Execute.Assertion
+                .ForCondition(subject != null && subject.HasFlag(expectedFlag))
+                .BecauseOf(because, reasonArgs)
+                .FailWith("The enum was expected to have flag {0} but found {1}{reason}.", expectedFlag, subject);
+
+            return new AndConstraint<ObjectAssertions>(this);
+        }
```

Comment:
So essentially something like so:
```csharp
Execute.Assertion
    .BecauseOf(because, reasonArgs)
    .ForCondition(Subject.GetType() == expectedFlag.GetType())
    .FailWith("some text")
    .Then
    .Given(() => Subject as Enum)
    .ForCondition(enum => enum.HasFlag(expectedFlag))
    .FailWith("some other text");
```